src/tests/testsuite.ts

```diff
@@ -124,9 +125,56 @@ export const createTestSuite: CreateTestsuite = (name, doubleCsrfOptions) => {
           : (mockRequest.cookies[cookieName] =
               (decodedCookieValue as string).split("|")[0] + "|invalid-hash");
 
-        expect(() => generateToken(mockRequest, mockResponse)).to.throw(
-          invalidCsrfTokenError.message
+        expect(() =>
+          generateToken(mockRequest, mockResponse, false, true)
+        ).to.throw(invalidCsrfTokenError.message);
+
+        // just an invalid value in the cookie
+        signed
+          ? (mockRequest.signedCookies[cookieName] = `s:${sign(
+              "invalid-value",
+              mockRequest.secret as string
+            )}`)
+          : (mockRequest.cookies[cookieName] = "invalid-value");
+
+        expect(() =>
+          generateToken(mockRequest, mockResponse, false, true)
+        ).to.throw(invalidCsrfTokenError.message);
+      });
+
+      it("should not throw (instead, overwrites wrong token) if csrf cookie is present, it is invalid (wrong token + hash pair, or not a correct value), overwrite is false, and validateOnGeneration is false", () => {
```

Comment:
should not throw if csrf cookie is present and invalid when overwrite is false, and validateOnGeneration is disabled